src/execution_context.cpp

```diff
@@ -203,21 +206,37 @@ bool IsHipKernelsEnabled()
 #endif
 }
 
-void miopen::ExecutionContext::DetectRocm()
+ExecutionContext& ExecutionContext::DetectRocm()
 {
     use_binaries            = false;
     use_asm_kernels         = false;
     use_hip_kernels         = IsHipKernelsEnabled();
-    use_opencl_convolutions = !miopen::IsDisabled(MIOPEN_DEBUG_OPENCL_CONVOLUTIONS{});
+    use_opencl_convolutions = !IsDisabled(MIOPEN_DEBUG_OPENCL_CONVOLUTIONS{});
     rmv                     = rocm_meta_version::Default;
     if(IsAmdRocmOpencl(*this))
     {
-        use_asm_kernels =
-            !miopen::IsDisabled(MIOPEN_DEBUG_GCN_ASM_KERNELS{}) && ValidateGcnAssembler();
+        use_asm_kernels = !IsDisabled(MIOPEN_DEBUG_GCN_ASM_KERNELS{}) && ValidateGcnAssembler();
 #ifndef HIP_OC_FINALIZER
-        use_binaries = !miopen::IsDisabled(MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES{});
+        use_binaries = !IsDisabled(MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES{});
 #endif
     }
+    return *this;
+}
+
+ExecutionContext& ExecutionContext::SetupFloats(const conv::ProblemDescription& problem)
```

Comment:
@DrizztDoUrden `ExecutionContext` is also used with activation, pooling and batchnorm solvers, and they have their own `ProblemDescription`

Reply:
Will move to `conv::ProblemDescription`